### PR TITLE
[Routing] Add fallback to cultureless locale for internationalized routes

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/issues/27938
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/10108

Please check the related issue for the context.

TODO : 
- [x] Update CHANGELOG
- [x] Update documentation

### CODE REVIEW

File: src/Symfony/Component/Routing/Tests/Generator/Dumper/PhpGeneratorDumperTest.php [hunk 1]
```diff
@@ -46,8 +46,8 @@ protected function setUp()
 
         $this->routeCollection = new RouteCollection();
         $this->generatorDumper = new PhpGeneratorDumper($this->routeCollection);
-        $this->testTmpFilepath = sys_get_temp_dir().DIRECTORY_SEPARATOR.'php_generator.'.$this->getName().'.php';
-        $this->largeTestTmpFilepath = sys_get_temp_dir().DIRECTORY_SEPARATOR.'php_generator.'.$this->getName().'.large.php';
+        $this->testTmpFilepath = sys_get_temp_dir().\DIRECTORY_SEPARATOR.'php_generator.'.$this->getName().'.php';
```
Comment (fancyweb) [inline src/Symfony/Component/Routing/Tests/Generator/Dumper/PhpGeneratorDumperTest.php]: This changes were applied by the fabbot patch. I guess I should revert them ?

**Review by nicolas-grekas — APPROVED** [review]
(just updated the implementation to make it shorter)

**Comment by fancyweb** [comment]
@nicolas-grekas Great job on making this shorter ! Is there any performance impact to "reprocess" everything everytime you generate an url ?

**Comment by nicolas-grekas** [comment]
> performance impact to "reprocess

the only impact this can have are at most two extra calls to strstr. I think that's OK :)

**Comment by nicolas-grekas** [comment]
(please fix fabbot issues)

**Comment by nicolas-grekas** [comment]
Thank you @fancyweb.

**Comment by ampaze** [comment]
Thanks from me as well.

**Pull request merged**